xrspatial/tests/conftest.py

```diff
@@ -0,0 +1,10 @@
+import pytest
+
+import numpy as np
+
+
+@pytest.fixture
+def random_data(size, dtype):
+    data = np.random.randint(low=-100, high=100, size=size)
```

Comment:
Use of the original numpy random interface is deprecated, it is now recommended to use the `default_rng` instead.

Also, all tests should be deterministic so random number generators should have a `seed`, even though in this case we are only using the numbers to compare different algorithms, e.g. `numpy` vs `cupy`. Although the test should pass regardless of the random numbers generated, in practice we want to avoid any test failure occurring at some point in the future as the failure will occur in an unrelated PR and will be unrepeatable and hence very difficult to track down.

So here we could have something like:
```python
rng = np.random.default_rng(2841)
data = rng.integers(-100, 100, size=size, dtype=dtype)
```

Reply:
Thanks Ian, this is a very good catch. As we use both int and float dtype for testing, I made a change that is slightly different. We'll convert to expected `dtype` after creating the array with `rng.integers()`